src/components/AddPeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -0,0 +1,280 @@
+      </v-btn>
+    </div>
+    <div class="btn-panel"  v-if="orgPersonList.length > 0">
+      <v-btn outlined color="primary" @click="addOrgPerson([], IncorporatorTypes.PERSON)"
+      :disabled="showOrgPersonForm" id="btn-add-person">
+        <v-icon>mdi-account-plus</v-icon>
+        <span>Add a Person</span>
+      </v-btn>
+      <v-btn outlined color="primary" :disabled="showOrgPersonForm" class="spacedButton"
+      @click="addOrgPerson([Roles.INCORPORATOR], IncorporatorTypes.CORPORATION)" id="btn-add-corp">
+        <v-icon>mdi-domain-plus</v-icon>
+        <span>Add a Corporation or Firm</span>
+      </v-btn>
+      <v-btn outlined color="primary" @click="addOrgPerson([Roles.COMPLETING_PARTY], IncorporatorTypes.PERSON)"
+      :disabled="showOrgPersonForm"  class="spacedButton" v-if="!hasRole(Roles.COMPLETING_PARTY, 1, 'ATLEAST')"
+      id="btn-add-cp">
+        <v-icon>mdi-account-plus-outline</v-icon>
+        <span>Add the Completing Party</span>
+      </v-btn>
+    </div>
+    <v-card flat class="people-roles-container" v-if="showOrgPersonForm">
+      <OrgPerson v-show="showOrgPersonForm"
+      :initialValue="currentOrgPerson"
+      :activeIndex="activeIndex"
+      :nextId="nextId"
+      :existingCompletingParty="completingParty"
+      @addEditPerson="onAddEditOrgPerson($event)"
+      @removePersonEvent="onRemovePerson($event)"
+      @resetEvent="resetData()"
+      @removeCompletingPartyRole="removeCompletingPartyAssignment()" />
+    </v-card>
+
+    <v-card flat v-if="orgPersonList.length > 0" :disabled="showOrgPersonForm" >
+      <ListPeopleAndRoles
+        v-if="orgPersonList.length > 0"
+        :personList="orgPersonList"
+        :isSummary="false"
+        @editPerson="editOrgPerson($event)"
+        @removePerson="onRemovePerson($event)"/>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+import { Action, State } from 'vuex-class'
+
+// Interfaces
+import { OrgPersonIF, BaseAddressObjIF, BaseAddressType, FormType, ActionBindingIF } from '@/interfaces'
+
+// Mixins
+import { EntityFilterMixin } from '@/mixins'
+
+// Enums
+import { EntityTypes, Roles, IncorporatorTypes } from '@/enums'
+
+// Components
+import OrgPerson from './OrgPerson.vue'
+import ListPeopleAndRoles from './ListPeopleAndRoles.vue'
+
+// Schemas
+import { addressSchema } from '@/schemas'
+
+@Component({
+  components: {
+    OrgPerson,
+    ListPeopleAndRoles
+  }
+})
+export default class PeopleAndRoles extends Mixins(EntityFilterMixin) {
+   $refs!: {
+    addPersonForm: FormType,
+    mailingAddressNew: BaseAddressType,
+    deliveryAddressNew: BaseAddressType
+  }
+
+   // State
+  @State(state => state.stateModel.addPeopleAndRoleStep.orgPeople)
+  readonly orgPersonList: OrgPersonIF[]
+
+  @Action setOrgPersonList!: ActionBindingIF
+  @Action setAddPeopleAndRoleStepValidity!: ActionBindingIF
+
+  private newOrgPerson: OrgPersonIF = {
```

Comment:
This is a good use case for a constructor. In TS in a previous project (albeit using a different framework) we defined "classes" that always had a constructor (for initial values) and a copy-constructor (to create a new object from an existing one).

I'm not sure if implementing a constructor is any cleaner for an interface... Maybe look into it quickly; if not then current code is OK.

Reply:
Or you could use a factory method to return a new, initialized object.

Reply:
I believe this is something we could refactor in the near future! If possible, since this is just a recommendation, can we leave this comment un-resolved so i can return to it post-PR?

Reply:
Yep, don't fix today. This was more of a observation although if there's a future opportunity to improve this then let's consider it.

Reply:
100% agree!